integration_tests/iprof_mpi_daemon_test/test.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Get base real-time signal number
+SIGRTMIN=$(kill -l SIGRTMIN)
+
+# Signal handler for capturing signals
+handle_signal() {
+    echo "Received signal $1 from mpi_daemon"
+}
+
+# Setup trap for ready signal sent from signal daemon
+trap 'handle_signal RT_SIGNAL_READY' $((SIGRTMIN + 0))
+
+# Get the PID of this script
+PARENT_PID=$$
+
+# Start mpi_daemon in the background
+./mpi_daemon $PARENT_PID &
+DAEMON_PID=$!
+
+# Allow the daemon to initialize
+sleep 2
+
```

Comment:
Please don't do that. This this racy and nighmarish in a super-slow CI environment:

In ruby I do something like
```

  def lazy_exec(&block)
    return unless in_mpi_env?

    Signal.trap(RT_SIGNAL_READY) do
      return
    end
    block.call
    sleep
  end
  ```
  
  Aka: way that the daemon send a signal `RT_SIGNAL_READY` to the PARENT_PID.

Reply:
Ah yes, will replace all sleeps with something similar.

Reply:
Added "wait for signal" function in 2f8d513.